src/frontend/apps/lti_site/components/App/AppContentLoader/index.tsx

```diff
@@ -73,17 +73,22 @@ const intl = createIntl(
   cache,
 );
 
-const appsContent: Record<string, LazyExoticComponent<ComponentType<any>>> = {};
+const appsContent: Record<string, LazyExoticComponent<ComponentType<any>>> = {
+  LTI: lazy(() => import('components/LTIRoutes')),
+  Site: lazy(() => import('components/SiteRoutes')),
+};
 Object.values(appNames).forEach((app) => {
   appsContent[app] = lazy(() => import(`apps/${app}/components/Routes`));
 });
-const ltiSite = lazy(() => import('components/LTIRoutes'));
 
 const AppContentLoader = () => {
   const appConfig = useAppConfig();
   const queryClient = useMemo(() => new QueryClient(), []);
 
-  const Content = appConfig.appName ? appsContent[appConfig.appName] : ltiSite;
+  let Content: LazyExoticComponent<ComponentType<any>>;
+  if (appConfig.appName) Content = appsContent[appConfig.appName];
+  else if (appConfig.frontend) Content = appsContent[appConfig.frontend];
+  else throw new Error('application and frontend are not properly set');
```

Comment:
Why are you not fallbacking anymore on the `lti` ? In you first version it was the case.

Reply:
Because it is not supposed to happen, if the backend does not provide `appName` nor `frontend` there is a development problem. IMHO the fallback may hide unexpected behavior. (In the original code, there was no fallback either and it would also throw an error. My first suggestion was based on the actual code not the original one)

Reply:
I agree with you :+1:

Reply:
This also explains why the error message is not translated, we are really not expecting this error to show up in production, only in development (otherwise it means there is a big problem with the backend).